model/configuration.py

```diff
@@ -769,3 +774,567 @@ def excluded_audio_data_sources(cls, _db):
         if value is None:
             value = cls.EXCLUDED_AUDIO_DATA_SOURCES_DEFAULT
         return value
+
+
+class ExternalIntegrationOwner(object):
+    """Interface allowing to get access to an external integration"""
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def external_integration(self, db):
+        """Returns an external integration owned by this object
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :return: External integration owned by this object
+        :rtype: core.model.configuration.ExternalIntegration
+        """
+        raise NotImplementedError()
+
+
+class CollectionExternalIntegrationOwner(object):
+    """Interface allowing to get access to an external integration"""
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def collection_external_integration(self, db, collection_id):
+        """Returns an external integration owned by the collection
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :param collection_id: Collection's ID
+        :type collection_id: int
+
+        :return: External integration owned by the collection
+        :rtype: core.model.configuration.ExternalIntegration
+        """
+        raise NotImplementedError()
+
+
+class BaseConfigurationStorage(object):
+    """Serializes and deserializes values as library's configuration settings"""
```

Comment:
Here and in `ConfigurationStorage` and `CollectionConfigurationStorage`, the term "library" is used in a way I think is inaccurate. A `ConfigurationSetting` _may_ be associated with an `ExternalIntegration` and a `Library`, but that's not what happens in any of these cases:

* This is an abstract class
* `ConfigurationStorage` is used to store settings on an `ExternalIntegration` alone
*  `CollectionConfigurationStorage` is used to store settings on an `ExternalIntegration` and a `Collection`.

Take out the term "library's"; it would only be appropriate in a class called `LibraryConfigurationStorage`.

Reply:
Changed it accordingly